Sources/DangerDependenciesResolver/PackageManager.swift

```diff
@@ -0,0 +1,232 @@
+
+        return package
+    }
+
+    private func save(package: Package) throws {
+        try fileCreator.createFile(atPath: folder.appendingPath(package.name), contents: package.encoded())
+    }
+
+    private func addMissingPackageFiles() throws {
+        for pinnedPackage in try packageDataProvider.resolvePinnedPackages(generatedFolder: generatedFolder) {
+            guard !folder.containsItem(named: pinnedPackage.name) else {
+                continue
+            }
+
+            let package = Package(
+                name: pinnedPackage.name,
+                url: pinnedPackage.url,
+                majorVersion: pinnedPackage.state.version.major
+            )
+
+            try save(package: package)
+        }
+    }
+
+    private func absoluteRepositoryURL(from url: URL) -> URL {
+        guard !url.isForRemoteRepository else {
+            return url
+        }
+
+        let path = url.absoluteString
+        return URL(string: path)!
+    }
+
+    func symlinkPackages(to folder: String) throws {
+        let checkoutsFolder = generatedFolder.appendingPath(".build/checkouts")
+        let repositoriesFolder = generatedFolder.appendingPath(".build/repositories")
+        let resolvedPackageFile = generatedFolder.appendingPath("Package.resolved")
+
+        guard FileManager.default.fileExists(atPath: checkoutsFolder),
+            FileManager.default.fileExists(atPath: repositoriesFolder) else {
+            try updatePackages()
+            return try symlinkPackages(to: folder)
+        }
+
+        let buildFolder = try folder.createSubfolderIfNeeded(withName: ".build")
+
+        if !buildFolder.containsItem(named: "checkouts") {
+            try buildFolder.createSymlink(to: checkoutsFolder, at: "checkouts")
+        }
+
+        if !buildFolder.containsItem(named: "repositories") {
+            try buildFolder.createSymlink(to: repositoriesFolder, at: "repositories")
+        }
+
+        if !folder.containsItem(named: "Package.resolved") {
+            try folder.createSymlink(to: resolvedPackageFile, at: "Package.resolved")
+        }
+    }
+
+    func makePackageDescription(for script: Script) throws -> Data {
+        guard let masterDescription = try? String(contentsOfFile: generatedFolder.appendingPath("Package.swift")) else {
+            try updatePackages()
+            return try makePackageDescription(for: script)
+        }
+
+        let toolsVersion = try resolveSwiftToolsVersion(executor: ShellExecutor(), onFolder: generatedFolder)
+        let expectedHeader = packageGenerator.makePackageDescriptionHeader(forSwiftToolsVersion: toolsVersion)
+
+        guard masterDescription.hasPrefix(expectedHeader) else {
+            try packageGenerator.generateMasterPackageDescription(forSwiftToolsVersion: toolsVersion)
+            return try makePackageDescription(for: script)
+        }
+
+        return masterDescription.replacingOccurrences(of: masterPackageName, with: script.name).data(using: .utf8) ?? Data()
```

Comment:
<!--
  0 failure: 
  1 warning:  Line should be 12...
  
  
  DangerID: danger-id-Danger;
  File: Sources/DangerDependenciesResolver/PackageManager.swift;
  Line: 146;
-->

- :warning: Line should be 120 characters or less: currently 124 characters (`line_length`)